packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -563,16 +637,25 @@ class OffsetLayer extends ContainerLayer {
     transform.multiply(Matrix4.translationValues(offset.dx, offset.dy, 0.0));
   }
 
+  /// Consider this layer as the root and build a scene (a tree of layers)
+  /// in the engine.
+  ui.Scene buildScene(ui.SceneBuilder builder) {
```

Comment:
why does this take a builder instead of creating it?

Reply:
Because `builder` could be customized by the caller: https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/rendering/layer.dart#L610